### ISSUE TITLE
Zabbix server role: zabbix_server_dbpassword_hash_method is never used

### ISSUE THREAD

**Issue by loricvdt** [ansible-collections/community.zabbix#1074]
##### SUMMARY
`zabbix_server_dbpassword_hash_method` is mentioned in the server role docs but is not present in the role tasks and therefore has no effect

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
Zabbix server role

##### ANSIBLE VERSION
```paste below
ansible [core 2.15.0]
  config file = /home/user/project/ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/user/.local/lib/python3.10/site-packages/ansible
  ansible collection location = /home/user/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0] (/usr/bin/python3)
  jinja version = 3.1.2
  libyaml = True
```
##### CONFIGURATION
```paste below
CONFIG_FILE() = /home/user/project/ansible.cfg
DEFAULT_JINJA2_EXTENSIONS(/home/user/project/ansible.cfg) = jinja2.ext.do
DEFAULT_REMOTE_USER(/home/user/project/ansible.cfg) = ansible
INTERPRETER_PYTHON(/home/user/project/ansible.cfg) = auto
```
##### OS / ENVIRONMENT / Zabbix Version
Zabbix Server 6.0
PostgreSQL database
Debian 12

##### STEPS TO REPRODUCE
```yaml
- name: Install Zabbix Server
  ansible.builtin.include_role:
    name: community.zabbix.zabbix_server
  vars:
    zabbix_server_version: 6.0
    zabbix_server_database: pgsql
    zabbix_server_dbpassword: dbpassword
    zabbix_server_dbpassword_hash_method: scram-sha-256
```
##### EXPECTED RESULTS

Password gets saved with the SCRAM-SHA-256 hash method in the PostgreSQL database

##### ACTUAL RESULTS

Password gets saved as MD5

```bash
psql -c "SELECT rolname, rolpassword FROM pg_authid WHERE rolname='zabbix-server'"
    rolname    |             rolpassword             
---------------+-------------------------------------
 zabbix-server | md5xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
```

**Comment by pravi** [comment]
I can confirm this and this makes psql fail since postgresql 14+ has scram-sha-256 as default (and allowed method in pg_hba.conf)

**Comment by pravi** [comment]
Also the playbook fail if user is already existing with scram-sha-256 passwords.

**Comment by pravi** [comment]
And zabbix web fails to connect to db as well

![image](https://github.com/ansible-collections/community.zabbix/assets/708222/3bd566d3-1c2a-46f1-8691-4fb2e8a8079a)

**Comment by krauthosting** [comment]
Seems we lost secure PostgreSQL password hashes since #886
Researched to be lost in those two big squash commits for 2.0.0:
https://github.com/ansible-collections/community.zabbix/pull/979
https://github.com/ansible-collections/community.zabbix/pull/904
The regression came from this also rather big commit likely unintentional:
https://github.com/ansible-collections/community.zabbix/pull/904/commits/70a9580608e9278aff297c412b1f3dd30ba40776#diff-649a94c0f23b0c25f85e71a995cce6713c24320b706fb9004495d77daa8beca1
@pyrodie18 @BGmot @dj-wasabi Let's fix it but maybe also reflect :hugs: 
Don't get why testing didn't catch it? MD5 isn't PG default anymore.

**Comment by pyrodie18** [comment]
Testing didn't catch it because md5 is still a valid option for PostgreSQL so it never failed.  Go back to the genisis of the ticket (`zabbix_server_dbpassword_hash_method` not actually being used).  I vaguely recall when I was making these changes cleaning up the role for 2.0.0 and if I remember correctly I removed the logic because as best as I could tell it was a leftover from previous edits.  Looking at history of the line in the docs (https://github.com/ansible-collections/community.zabbix/blame/2d72df018116e99a606b07ae47b12cd8a511e60c/docs/ZABBIX_SERVER_ROLE.md#L137), it looks like you added the line to the documentation 3 months before we did the release of 2.0.0 but I believe (although not positive) we were well on the way to doing the rework for 2.0.0 at that time so I suspect we crossed over each other's work without realizing it.  Because there weren't any conflicts, it never grabbed anyone's attention to resolve so no one took a closer look at it.
